rllib/agents/trainer.py

```diff
@@ -1432,30 +1432,18 @@ def training_iteration(self) -> ResultDict:
         Returns:
             The results dict from executing the training iteration.
         """
-        # Some shortcuts.
-        batch_size = self.config["train_batch_size"]
-
-        # Collects SampleBatches in parallel and synchronously
-        # from the Trainer's RolloutWorkers until we hit the
-        # configured `train_batch_size`.
-        sample_batches = []
-        num_env_steps = 0
-        num_agent_steps = 0
-        while (not self._by_agent_steps and num_env_steps < batch_size) or (
-            self._by_agent_steps and num_agent_steps < batch_size
-        ):
-            new_sample_batches = synchronous_parallel_sample(self.workers)
-            sample_batches.extend(new_sample_batches)
-            num_env_steps += sum(len(s) for s in new_sample_batches)
-            num_agent_steps += sum(
-                len(s) if isinstance(s, SampleBatch) else s.agent_steps()
-                for s in new_sample_batches
+        # Collect SampleBatches from sample workers until we have a full batch.
+        if self._by_agent_steps:
+            train_batch = synchronous_parallel_sample(
+                worker_set=self.workers, max_agent_steps=self.config["train_batch_size"]
```

Comment:
So I guess we are rolling out the max_agent_steps/max_env_steps thing with this PR? (Related comment above)